File: BOJ/Silver/11399.py
# [BOJ] 11399. ATM

# 각 사람이 돈을 인출하는데 필요한 시간의 합의 최솟값

N = int(input())  # 사람의 수
times = list(map(int, input().split()))  # 각 사람이 돈을 인출하는데 걸리는 시간

times.sort()  # 작은 순서대로 정렬
ans = 0  # 정답 변수를 0으로 만들기

for i in range(1, N+1):
    ans += sum(times[0:i])  # 리스트의 0 ~ i까지 시간 더하기
print(ans)

File: BOJ/Silver/11399.js
// [BOJ] 11399. ATM

const fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const N = Number(input.shift()); // 사람의 수

// 각 사람이 돈을 인출하는데 걸리는 시간 오름차순으로 정렬 후
// 남은 사람들 수 만큼 곱하여 누적 합 구하기
const result = input[0]
  .split(" ")
  .map(Number)
  .sort((a, b) => a - b)
  .reduce((acc, cur, idx) => acc + cur * (N - idx), 0);

console.log(result);
